test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -465,6 +465,11 @@ object Assertion {
 
   /**
    * Makes an assertion that requires a value have the specified type.
+   *
+   * Example:
+   * {{{
+   *   assert(Duration.fromNanos(1), isSubtype[Duration.Finite](Assertion.anything))
```

Comment:
Great additional documentation! I wonder if we should add a `isSubtypeOf` assertion that doesn't take another parameter to make this use case more ergonomic.

Reply:
I was really confused about that when I first started coding it - to be honest this was the only odd thing in the whole PR after I've read the ZIO test spec. So as a new user of this I'd recommend having an option to call isSubtype[A]()